components/test-suite/src/loci/tests/testng/FormatReaderTest.java

```diff
@@ -963,10 +965,11 @@ public void testPhysicalSizeZ() {
         expectedSize = null;
       }
       Length realSize = retrieve.getPixelsPhysicalSizeZ(i);
-      Number size = realSize == null ? null : realSize.value(UNITS.MICROM);
+      Double size = realSize == null ? null : realSize.value(UNITS.MICROM).doubleValue();
 
       if (!(expectedSize == null && size == null) &&
-        (expectedSize == null || !expectedSize.equals(size)))
+        (expectedSize == null || (
+          Math.abs(size - expectedSize) > Constants.EPSILON)))
       {
```

Comment:
I think there might need to be one additional set of null checks for each of these three cases.  ```realSize.value(UNITS.MICROM)``` can be null even if ```realSize``` is non-null; I suspect this is what's behind the 10 unlogged failures in https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-5.1-merge-full-repository/558/ (2 in the ```imspector``` directory, not sure yet which directory the other 8 are from).